src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -87,7 +88,7 @@ internal abstract partial class AnalyzerDriver : IDisposable
         /// <summary>
         /// Lazily populated dictionary indicating whether a source file has any hidden regions - we populate it lazily to avoid realizing all syntax trees in the compilation upfront.
         /// </summary>
-        private Dictionary<SyntaxTree, bool> _lazyTreesWithHiddenRegionsMap;
+        private ConcurrentDictionary<SyntaxTree, bool> _lazyTreesWithHiddenRegionsMap;
```

Comment:
Did you give consideration to merging these dictionaries? Seems like we'd be a lot more efficient allocation wise if we had a single dictionary `ConcurrentDictionary<SyntaxTree, (bool?, bool?)>`

Reply:
I did not. Want me to file a follow up issue or just let it show up in a future allocation test if it's relevant?

Reply:
Lets wait and see if it shows up. My suspsicion is just that suspicions. Just seems wasteful here to have 3 maps where the keys are the same.